### Carregamento de bibliotecas

No bloco de código abaixo é feito o carregamento de todas as bibliotecas que serão utilizadas neste notebook.

In [2]:
from scipy.io import loadmat
import numpy as np

### Carregamento

No bloco de código abaixo, realizamos o carregamento de todos os pedaços da base de dados avi em `multi-channel`. 

É efetuada uma iteração entre 1 e 5 e concatenados os dados carregados na variável `avi_data`.

In [3]:
avi_data = list()

for i in range(1,5):
    avi_data.append(loadmat(f"../../datasets/avi/multi/Sub{i}_1_multitarget.mat"))
    avi_data.append(loadmat(f"../../datasets/avi/multi/Sub{i}_2_multitarget.mat"))

### Captura dos Dados

No trecho de código abaixo é executada a captura dos dados (X) e suas respectivas classes (y).

In [4]:
X = [d['Data'][0][0][7] for d in avi_data]
y = [d['Data'][0][0][8][0] for d in avi_data]

print("Divisão dos dados:")
for data in X:
    print(data.shape)

print("\nDivisão dos rótulos (sequências):")
for data in y:
    print(data)

Divisão dos dados:
(8192, 10)
(8192, 10)
(8192, 10)
(8192, 10)
(8192, 10)
(8192, 10)
(8192, 10)
(8192, 10)

Divisão dos rótulos (sequências):
[7.5 8.2 7.  8.2 6.  7.5 6.  6.  8.2 8.2]
[9.3 6.  6.  8.2 6.  6.5 7.5 7.  6.  6. ]
[7.5 8.2 7.  8.2 6.  7.5 6.  6.  8.2 8.2]
[9.3 6.  6.  8.2 6.  6.5 7.5 7.  6.  6. ]
[7.5 8.2 7.  8.2 6.  7.5 6.  6.  8.2 8.2]
[9.3 6.  6.  8.2 6.  6.5 7.5 7.  6.  6. ]
[7.5 8.2 7.  8.2 6.  7.5 6.  6.  8.2 8.2]
[9.3 6.  6.  8.2 6.  6.5 7.5 7.  6.  6. ]


### Organização dos dados

No trecho de código abaixo é realizado o `reshape` dos dados presentes na variável `X`.

O algoritmo realizará a seguinte transformação:

De

- (4, 8192, 10)

Para

- (4, 10, 8192)

In [5]:
reshape_X = np.array([np.transpose(data) for data in X])
reshape_X.shape

(8, 10, 8192)

### Salvando dados em um arquivo

No trecho de código abaixo é feito o salvamento dos dados formatados na extensão `.npy`.

In [6]:
np.save("../../datasets/avi/multi/data_multi.npy", reshape_X)
np.save("../../datasets/avi/multi/labels_multi.npy", y)